src/pages/blog/neon.md

```diff
@@ -0,0 +1,495 @@
+```rb
+# Be sure to restart your server when you modify this file.
+
+# Configure parameters to be partially matched (e.g. passw matches password) and filtered from the log file.
+# Use this to limit dissemination of sensitive information.
+# See the ActiveSupport::ParameterFilter documentation for supported notations and behaviors.
+Rails.application.config.filter_parameters += [
+  ...,
+  :database_information,
+]
+```
+
+So now we have our users, but what about our notes? There are several advantages here to a multi-tenant architecture:
+
+- **Security by Design:** There is significantly less risk of harm due to accidentally destroying/fetching other customers data. This is because instead of worrying about multiple columns when you do things to a users notes, you just have to think about one. Additionally, you don't have to worry about the risk of someone finding another users notes by decrementing their own note ID since the users notes are pulled from the signed in record.
+- **Data Compliance:** If in the future we wanted to deploy in a region that the user specifies for compliance reasons (for example, an American user might want to deploy in America if you have a point of presence there due to latency reasons, but a European user might want to deploy in Europe due to Neon's GDPR compliance), this is a drop down box away with this architecture. Additionally, data compliance is significantly easier here. Want to add GDPR exports? Just dump out the entire database. Want to delete a user without worrying about errors due to foreign keys within your large database? Just delete the database (and we will tie this to the User record in this tutorial).
+- **Competitive Advantage:** With the multi-database model, you don't have to worry about leaking information that whilst it may not be sensitive to your users, is quite revealing about your company. For example, in this case, you might be able to figure out the companies user retention by taking a publicly sent customer count and then comparing it to the note ID since it is auto-incrementing with ActiveRecord.
+
+The main tradeoff with this model is that migrations become more complex, although this can be managed with careful branching and scripting.
+
+To do this, we are going to need to make a few network requests to the Neon API. Since `net/http` is capped to HTTP/1.1, we are going to use `httpx` to do this faster. Add this to your `Gemfile` and then run `bundle`:
+
+```rb
+# Add support for higher than HTTP/1.1 clients.
+gem 'httpx', '~> 1.3.1'
+```
+
+To do this, we want to create a database and tie it to this record until it is destroyed. In case we later want to make a database per another entity (for example, one per organisation), we will make this a concern that we can share across anything that needs it. To do this, we will remove `app/models/concerns/.keep` and create a `multi_tenant.rb` file there to manage this:
```

Comment:
```suggestion
We want to create a database and tie it to this record until it is destroyed. In case we later want to make a database per another entity (for example, one per organisation), we will make this a concern that we can share across anything that needs it. To do this, we will remove `app/models/concerns/.keep` and create a `multi_tenant.rb` file there to manage this:
```